driver-core/src/main/java/com/datastax/driver/core/AbstractAddressableByIndexData.java

```diff
@@ -43,75 +48,111 @@ protected ByteBuffer getValue(int i) {
     }
 
     public T setBool(int i, boolean v) {
-        checkType(i, DataType.Name.BOOLEAN);
-        return setValue(i, TypeCodec.BooleanCodec.instance.serializeNoBoxing(v));
+        checkType(i, BOOLEAN);
+        TypeCodec<Boolean> codec = codecFor(i, Boolean.class);
+        ByteBuffer bb;
+        if(codec instanceof TypeCodec.PrimitiveBooleanCodec)
+            bb = ((TypeCodec.PrimitiveBooleanCodec)codec).serializeNoBoxing(v, protocolVersion);
```

Comment:
serialize calls serializeNoBoxing and deserialize calls deserializeNoBoxing. The code you see is an attempt to preserve a feature that existed in the previous version: avoid unnecessary boxings/unboxings of primitive types. Since we cannot guarantee anymore that the codec that will be used to serialize a primitive boolean is BooleanCodec, we now test if the returned codec implements PrimitiveBooleanCodec, in which case the special method serializeNoBoxing is called, thus avoiding the boxing/unboxing overhead (and obviously, BooleanCodec now implements PrimtivieBooleanCodec).